auth/auth-router.js

```diff
@@ -1,7 +1,44 @@
 const router = require("express").Router();
+const bcrypt = require("bcryptjs");
 
-router.post("/register", (req, res) => {});
+const users = require("./auth-model");
+const signToken = require("./auth-mw/signToken");
+const validateFields = require("./auth-mw/validateUserFields");
+const constants = require("../variables/constants");
+const requireEmail = require("./auth-mw/requireEmail");
 
-router.post("/login", (req, res) => {});
+router.post("/register", validateFields, requireEmail, (req, res) => {
+  const user = req.body;
+
+  const hash = bcrypt.hashSync(user.password, constants.ROUNDS);
+
+  user.password = hash;
+
+  users
+    .add(user)
+    .then((id) => {
+      res.status(201).json({ message: "User successfully created", id });
+    })
+    .catch((err) => {
+      res.status(500).json({ message: err });
+    });
+});
+
+router.post("/login", validateFields, (req, res) => {
+  const { username, password } = req.body;
+
+  users.selectBy({ username: username }).then(([user]) => {
+    if (user && bcrypt.compareSync(password, user.password)) {
+      const token = signToken(user);
+
+      res.json({
+        message: "Login successful",
+        token,
+      });
+    } else {
+      res.status(401).json({ error: "invalid login attempt" });
+    }
+  });
+});
```

Comment:
Clean formatting 
basic Boilerplate
errors for server vs client side errors are present.
everything here looks in place 🚀